solana/program_v2/programs/gateway_v2/src/instructions/network/close_gatekeeper.rs

```diff
@@ -0,0 +1,50 @@
+use crate::constants::GATEKEEPER_SEED;
+use crate::state::*;
+use anchor_lang::prelude::*;
+
+pub fn close_gatekeeper(ctx: Context<CloseGatekeeperAccount>) -> Result<()> {
+    let network = &mut ctx.accounts.network;
+    let gatekeeper = &mut ctx.accounts.gatekeeper;
+
+    // Find and remove the gatekeeper from the network
+    let index = network
+        .gatekeepers
+        .iter()
+        .position(|k| k == &gatekeeper.key());
+
+    if let Some(key_index) = index {
+        network.gatekeepers.remove(key_index);
+    }
```

Comment:
Question: Should we throw/return Error when not found? I highlighted these errors in did:sol.